src/hooks/useManifestState/reducer.js

```diff
@@ -62,6 +62,16 @@ const setRows = (state, action) => {
   })
 }
 
+const decrementPage = ({ rows, page }) => {
+  if (rows.length) {
+    return page
+  }
+  if (page) {
+    return page - 1
+  }
+  return 0
+}
```

Comment:
Can this be replaced with Math.min?

Reply:
This might still have the infinite loop bug.  Test with a console log in the correctRowCount function to verify

Reply:
It wasn't looping, correctRowCount only ever got called once per page render, I didn't see a way of doing this with Math.min() I assumed you were thinking of Math.max() instead which cleaned this up a bit, the problem was that it was returning negative page numbers so ```Math.max(page - 1, 0)``` checks for negatives

Reply:
Yes, I always get that backwards! :facepalm: